src/pynwb/legacy/map.py

```diff
@@ -25,54 +25,93 @@ class TypeMapLegacy(TypeMap):
 
     def get_builder_dt(self, builder):
 
-        if builder.name == 'roi_ids':
-            pass
-        elif builder.name == 'root':
+        if builder.name == 'root':
             return 'NWBFile'
-        attrs = builder.attributes
-        ndt = attrs.get('neurodata_type')
-        if ndt == 'Module':
-            return 'ProcessingModule'
-        elif ndt == 'TimeSeries':
-            ancestry = attrs['ancestry']
-            if decode(ancestry[-1]) == 'TwoPhotonSeries' and decode(builder.name) == 'corrected':
-                return 'ImageSeries'
-            else:
-                return decode(ancestry[-1])
-
-        elif ndt == 'Interface':
-            return builder.name
-        if ndt == 'Epoch':
-            return 'Epoch'
         else:
-            parent_ndt = self.get_builder_dt(builder.parent)
-            if parent_ndt == 'Epoch':
-                return 'EpochTimeSeries'
-            if parent_ndt == 'MotionCorrection':
-                return 'CorrectedImageStack'
-            if parent_ndt == 'ImagingPlane' and isinstance(builder, GroupBuilder):
-                return 'OpticalChannel'
-            if parent_ndt == 'ImageSegmentation':
-                if builder.name in ('roi_ids', 'cell_specimen_ids'):
+
+            attrs = builder.attributes
+            ndt = attrs.get('neurodata_type')
+
+            if ndt == 'Module':
```

Comment:
To reduce the complexity, consider adding a function named `_get_builder_dt_from_neurodata_type(neurodata_type, attrs) ?`

Reply:
I want to keep the functionality as parallel to src/form/build/map.py::get_builder_dt, so I would prefer to use noqa: C901 unless you really think it is unreable w/o a refactor.